sys-libs/compiler-rt/compiler-rt-11.0.0.ebuild

```diff
@@ -83,8 +87,13 @@ src_configure() {
 
 	if use prefix && [[ "${CHOST}" == *-darwin* ]] ; then
 		mycmakeargs+=(
-			# disable use of SDK for the system itself
-			-DDARWIN_macosx_CACHED_SYSROOT=/
+			# setting -isysroot is disabled with compiler-rt-prefix-paths.patch
+			# this allows adding arm64 support using SDK in EPREFIX
+			-DDARWIN_macosx_CACHED_SYSROOT="${EPREFIX}/MacOSX.sdk"
+			# Set this based on the SDK in EPREFIX
+			-DDARWIN_macosx_OVERRIDE_SDK_VERSION="$(realpath ${EPREFIX}/MacOSX.sdk | sed -e 's/.*MacOSX\(.*\)\.sdk/\1/')"
```

Comment:
Is it necessary to hardcode the symlink target here?  This way, we cannot switch the symlink (e.g. using eselect, or manually after Xcode update).  Is this used only during build, or also from the library itself during runtime?

Reply:
The SDK version number is not baked into the build.
It does however disable i386 for SDK >= 10.15, and issue a fatal error building the TSan lib with SDK < 10.12.

in `compiler-rt/cmake/Modules/CompilerRTDarwinUtils.cmake`:
- function `find_darwin_sdk_version` is the only thing that uses this var
- in `darwin_test_archs`, it has this comment where the `find_darwin_sdk_version` function is used:
```
# Disable building for i386 for macOS SDK >= 10.15. The SDK doesn't support
# linking for i386 and the corresponding OS doesn't allow running macOS i386
# binaries.
```

in `lib/tsan/CMakeLists.txt`, it has this comment where `find_darwin_sdk_version` is used:
```
# Ideally we would check the SDK version for the actual platform we are
# building for here.  To make our lifes easier we assume the host SDK setup is
# sane and use the macOS SDK version as a proxy for aligned SDKs.
```
and then it throws this FATAL ERROR message for `VERSION_LESS 10.12`
```
message(FATAL_ERROR "Building the TSan runtime requires at least macOS SDK 10.12 (or aligned SDK on other platforms)")
```

Reply:
Also, I want to clarify the realpath usage: Here we get the realpath only to extract the version number from it with sed.
```
realpath ${EPREFIX}/MacOSX.sdk | sed -e 's/.*MacOSX\(.*\)\.sdk/\1/'
```

I considered reading `MacOSX.sdk/SDKSettings.json` but then I'd have to have a bdep on `jq` or use string magic and hope it doesn't break or something.

Reply:
I updated the comment to clarify what this is used for in the build.